tmt/steps/execute/__init__.py

```diff
@@ -1031,12 +1032,15 @@ def _run_checks_for_test(
         *,
         event: CheckEvent,
         invocation: TestInvocation,
+        internal: bool = False,
```

Comment:
Hm, once the method needs a hint like this, it's usually better to parametrize the input completely, i.e. instead of `internal: bool` to tell it which attribute to use, give it `checks: Sequence[Check]` or something, and call it with the actual list of checks to run. `run_checks_before_test`, `run_checks_after_test` and `run_internal_checks` can all pick the right list and pass it to this helper. It will make its API cleaner, no side-effect hidden inside.